indexing-service/src/main/java/io/druid/indexing/overlord/ThreadPoolTaskRunner.java

```diff
@@ -335,6 +379,11 @@ public TaskStatus call()
 
       try {
         log.info("Running task: %s", task.getId());
+        TaskRunnerUtils.notifyLocationChanged(
```

Comment:
BTW in general a task's location can be announced before it's really ready to listen. I think that's OK.